http_client.go

```diff
@@ -305,27 +321,41 @@ func (data *Data) Sign(reqPath string) (api string) {
 }
 
 func (data *Data) Send(url string) (Result, error) {
-	var result Result
 	req, err := http.NewRequest("POST", url, bytes.NewBuffer(data.dataBytes))
 	if err != nil {
-		return result, err
+		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
 	client := &http.Client{}
 	resp, err := client.Do(req)
 	if err != nil {
-		return result, err
+		return nil, err
 	}
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return result, err
+		return nil, err
 	}
+	var result response
 	err = json.Unmarshal(body, &result)
 	if err != nil {
-		return result, err
+		if resp.StatusCode != http.StatusOK {
+			return nil, newAPIError("HTTP " + resp.Status)
+		} else {
```

Comment:
`golint` 344:10 if block ends with a return statement, so drop this else and outdent its block